disorder-core/src/Disorder/Core/Gen.hs

```diff
@@ -15,3 +17,10 @@ vectorOfSize min' max' gen =
 chooseSize :: Int -> Int -> Gen Int
 chooseSize min' max' =
   sized (return . min max' . max min')
+
+-- | from a generator return a generator that will generate Nothing
+--   half the time and Just half the time
+maybeGen :: Gen a -> Gen (Maybe a)
+maybeGen g = do
+  b <- elements [True, False]
```

Comment:
oh, no, silly me - if you were using Maybe recursively, the resize should go there, not in Maybe. never mind!